File: DSAlgorithms/Leetcode/c++/sliding_window/3-longest_substring_without_repeats.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       std::unordered_set<char> a_set;
       int left = 0;
       int result = 0;
       for(int right = 0; right < s.size(); right++) 
       {
           std::unordered_set<char>::iterator char_iterator = a_set.find(s[right]);
           while(char_iterator != a_set.end())
           {
               a_set.erase(s[left]);
               left++;
              char_iterator = a_set.find(s[right]);
           }
           a_set.insert(s[right]);
           result = std::max(result, right - left + 1);
       }
       return result;
    }
};

File: dsa_algo/leetcode/ts/src/sliding_window/3-longest_substring_without_repeats.ts
const lengthOfLongestSubstring = (s: string): number =>{
    const a_set = new Set<string>();
    let left = 0;
    let result = 0;
    for(let right = 0; right < s.length; right++)
    {
        while(a_set.has(s[right]))
        {
            a_set.delete(s[left]);
            left++;
        }
        a_set.add(s[right]);
        result = Math.max(result, right - left + 1);
    }
    return result;
};